### PR TITLE
bump leaflet, esri leaflet, geocoder

### PR DESCRIPTION
**Pull request by jgravois**
i went on a bit of a tangent trying to debug ArcGIS/arcgis-for-developers#3178

### fixed
* changed default basemap to `Topographic`
* esri-leaflet and esri-leaflet-geocoder bumped to `1.x`
* moved the above, and `geotrigger-js` from git `submodules` to ~~`devDependencies`~~ CDN
* moved basemap control on top of geosearch so suggestions would stop obscuring the other control
* updated doc and changelog
* fixed `package.json` license tag to prevent `npm install` nag
* got grunt build etc. working again
* fixed some broken relative css `img` paths

### q's
* ~~i'm just assuming npm dependencies are nicer than submodules.  happy to retool if i'm wrong.~~
* ~~the `geotrigger-editor.css` got updated on its own.  i assume its because a newer version of one of the css build tools does its thing a touch differently.~~
* ~~calling `grunt dev` doesn't actually watch and rebuild when changes are made in `src/`.  (i didn't do anything about that).~~
* ~~calling `grunt build` works fine, but throws a warning about the code in `geotrigger-js`.  (i didnt do anything about that either).~~
* ~~`.png` resources aren't loaded in the `example/default/` (i didn't do anything about that either)~~

i sorted all q's after taking a fresh look at the code

### CODE REVIEW

**Comment by ngoldman** [comment]
my sincere apologies, and best of luck

**Comment by jgravois** [comment]
what's there to apologize for friend?

**Comment by ngoldman** [comment]
mistakes made long ago

**Comment by jgravois** [comment]
as per @paulcpederson and my conversation:
* moved everything possible to CDN and got rid of both submodules and npm dependencies
* bumped `grunt-contrib-uglify` to actually get the build working
* caught a broken relative path in the CSS (this fixed the /examples/
* admitted that i don't have enough knowledge or interest to debug whats wrong with the proxy sample
* bumped the changelog, as i think merging this definitely warrants tagging a release.

after everyone's happy, i'll port into the developer site.

File: package.json [hunk 1]
```diff
@@ -53,7 +55,11 @@
     {
       "name": "Jerry Sievert",
       "email": "code@legitimatesounding.com"
+    },
+    {
+      "name": "John Gravois",
```
Comment (paulcpederson) [inline package.json]: nice.

File: README.md [hunk 2]
```diff
@@ -133,14 +133,14 @@ To run the development environment, you will need the following:
 
 To install `node` on your system follow [these instructions](https://github.com/joyent/node/wiki/Installation#installing-without-building).
 
-*Note: if you're a mac developer I recommend simply using homebrew to `brew install node`.*
+*Note: if you're a mac developer we recommend using homebrew to `brew install node`.*
```
Comment (ngoldman) [inline README.md]: No longer recommended! Just download the installer from https://nodejs.org.

Comment (jgravois) [inline README.md]: thats why its *we*, we're smarter when we include you.

**Pull request merged**